proto/babylon/finality/v1/tx.proto

```diff
@@ -73,17 +76,30 @@ message MsgAddFinalitySigResponse{}
 // MsgUpdateParams defines a message for updating finality module parameters.
 message MsgUpdateParams {
     option (cosmos.msg.v1.signer) = "authority";
-  
+
     // authority is the address of the governance account.
     // just FYI: cosmos.AddressString marks that this field should use type alias
     // for AddressString instead of string, but the functionality is not yet implemented
     // in cosmos-proto
     string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
-  
+
     // params defines the finality parameters to update.
     //
     // NOTE: All parameters must be supplied.
     Params params = 2 [(gogoproto.nullable) = false];
 }
 // MsgUpdateParamsResponse is the response to the MsgUpdateParams message.
 message MsgUpdateParamsResponse {}
+
+// MsgUnjailFinalityProvider defines the Msg/UnjailFinalityProvider request type
+message MsgUnjailFinalityProvider {
+    option (gogoproto.goproto_getters) = false;
+    option (cosmos.msg.v1.signer) = "signer";
+
+    string signer = 1;
+    // fp_btc_pk is the BTC PK of the finality provider that commits the public randomness
+    bytes fp_btc_pk = 2 [ (gogoproto.customtype) = "github.com/babylonlabs-io/babylon/types.BIP340PubKey" ];
+}
```

Comment:
It seems that any BBN account (not just the BBN account that creates the FP) can unjail an arbitrary FP. Is this expected or we want to enforce that only the BBN account that creates the FP can unjail itself?

Reply:
OK thinking this further I think some authentication is beneficial here. Otherwise there is an attack: let's say a FP is jailed and the FP operator is not aware of that. Then an adversary keeps unjailing it and the protocol keeps jailing it, such that all rewards accumulated by this FP are slashed, before the FP operator realises that its FP is down. If there is an authentication then this attack won't happen.

The authentication is quite simple: get FP from DB, and compare whether the `Addr` of FP is same as `req.Signer`

Reply:
Nice point! We should definitely have the authentication. Maybe not for the monetary reason but because unplanned unjailing will ruin the fp's reputation.